### PR TITLE
fix: Regenerate device id function added

### PR DESCRIPTION
**Pull request by yuhao900914**

### CODE REVIEW

File: android/src/main/kotlin/com/amplitude/amplitude_flutter/AmplitudeFlutterPlugin.kt [hunk 1]
```diff
@@ -105,6 +105,11 @@ class AmplitudeFlutterPlugin : FlutterPlugin, MethodCallHandler {
 
                 result.success("setServerUrl called..")
             }
+            "regenerateDeviceId" -> {
+                val client = Amplitude.getInstance(instanceName)
+                client.regenerateDeviceId();
+                result.success(" regenerateDeviceId called..")
```
Comment (haoliu-amp) [inline android/src/main/kotlin/com/amplitude/amplitude_flutter/AmplitudeFlutterPlugin.kt]: Extra space in the string

File: example/ios/Flutter/Debug.xcconfig [hunk 2]
```diff
@@ -1,2 +1,3 @@
+#include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"
```
Comment (haoliu-amp) [inline example/ios/Flutter/Debug.xcconfig]: What is this? Don't feel it's needed. Same for below.

File: example/lib/regenerate_device.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+
+import 'app_state.dart';
+
+/*
+class RegenerateDeviceBtn extends StatefulWidget {
```
Comment (haoliu-amp) [inline example/lib/regenerate_device.dart]: Remove all these?

File: lib/amplitude.dart [hunk 4]
```diff
@@ -96,12 +100,18 @@ class Amplitude extends _Amplitude {
...
+        'regenerateDeviceId', jsonEncode(_baseProperties()));
+  }
+
   /// Dynamically adjust server URL
   Future<void> setUseDynamicConfig(bool useDynamicConfig) async {
     Map<String, dynamic> properties = _baseProperties();
     properties['useDynamicConfig'] = useDynamicConfig;
 
-    return await _channel.invokeMethod('setUseDynamicConfig', jsonEncode(properties));
+    return await _channel.invokeMethod(
```
Comment (haoliu-amp) [inline lib/amplitude.dart:117]: Try not to mix auto formatting the files together with your feature change. Usually you can create a separate PR for only doing formatting. It's fine to leave it this time.

File: lib/amplitude.dart [hunk 5]
```diff
@@ -96,12 +100,18 @@ class Amplitude extends _Amplitude {
     return await _channel.invokeMethod('setServerUrl', jsonEncode(properties));
   }
 
+  Future<void> regenerateDeviceId() async {
```
Comment (haoliu-amp) [inline lib/amplitude.dart:107]: Add comments /// for better documentation.

File: ios/Classes/SwiftAmplitudeFlutterPlugin.swift [hunk 6]
```diff
@@ -82,8 +82,10 @@ import Amplitude
                     }
 
                     result(true)
-
                 // Event logging
```
Comment (haoliu-amp) [inline ios/Classes/SwiftAmplitudeFlutterPlugin.swift]: Comment is missing for `regenerateDeviceId`

File: ios/Classes/SwiftAmplitudeFlutterPlugin.swift [hunk 7]
```diff
@@ -82,8 +82,10 @@ import Amplitude
...
+                case "regenerateDeviceId":
+                    Amplitude.instance(withName: instanceName).regenerateDeviceId()
+                    result(true)
                 case "logEvent":
```
Comment (haoliu-amp) [inline ios/Classes/SwiftAmplitudeFlutterPlugin.swift:90]: Where's the original comment?

**Review by haoliu-amp — CHANGES_REQUESTED** [review]

File: lib/amplitude.dart [hunk 8]
```diff
@@ -96,12 +100,22 @@ class Amplitude extends _Amplitude {
     return await _channel.invokeMethod('setServerUrl', jsonEncode(properties));
   }
 
+  ///Regenerates a new random deviceId for current user.
```
Comment (haoliu-amp) [inline lib/amplitude.dart]: Space after /

**Review by haoliu-amp — CHANGES_REQUESTED** [review]

**Comment by yuhao900914** [comment]
Make a new pull request for correcting the request title.

**Pull request closed without merging**